internal/olm/operator/internal/configmap.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/base32"
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/internal/util/k8sutil"
+
+	"github.com/ghodss/yaml"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// RegistryDataStale checks if manifest data stored in the registry is stale
+// by comparing it to manifest data currently managed by m.
+func (m *RegistryResources) RegistryDataStale(ctx context.Context, namespace string) (bool, error) {
+	pkg := m.Manifests.GetPackageManifest()
+	pkgName := pkg.PackageName
+	nn := types.NamespacedName{
+		Name:      getRegistryConfigMapName(pkgName),
+		Namespace: namespace,
+	}
+	configmap := corev1.ConfigMap{}
+	err := m.Client.KubeClient.Get(ctx, nn, &configmap)
+	if err != nil {
+		return false, err
+	}
+	// Collect digests of manifests submitted to m.
+	newData, err := createConfigMapBinaryData(pkg, m.Manifests.GetBundles())
+	if err != nil {
+		return false, errors.Wrap(err, "error creating binary data")
+	}
+	// If lengths don't match we have added or removed a file.
```

Comment:
Added /remove where? Wydt about make the comment more clear?

Reply:
I will update the comment to something more explanatory. I'm comparing the number of files in the new registry to the number of files in the existing registry. If the new number is different from the current number of files, we have either added or removed files, therefore the existing registry data is stale.